skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillModel.java

```diff
@@ -5,16 +5,16 @@
 import com.fasterxml.jackson.annotation.JsonBackReference;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import jakarta.persistence.*;
-import lombok.Data;
-import lombok.EqualsAndHashCode;
-import lombok.NoArgsConstructor;
+import lombok.*;
 
 import java.time.Instant;
 import java.util.Set;
 import java.util.UUID;
 
 @EqualsAndHashCode(callSuper = true)
 @Entity
+@Builder
+@AllArgsConstructor
 @NoArgsConstructor
 @Data
```

Comment:
It's best not to use `@Data` on Entity classes. The Equals and HashCode method is messed up in case you have foreign-key based collections.

Reply:
> The Equals and HashCode method is messed up in case you have foreign-key based collections.

Hi @debanjanc01 just curious to know more about it, can you share some article which you might have followed, I found this one https://stackoverflow.com/questions/27581/what-issues-should-be-considered-when-overriding-equals-and-hashcode-in-java

Reply:
Here are two links that can help you understand
- https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/
- https://thorben-janssen.com/lombok-hibernate-how-to-avoid-common-pitfalls/#dont-use-equalsandhashcode

You can try writing it out yourself in a sample app, and do a Get operation.
(ps. been there, done that, learned from experience)

Reply:
I was curious to expand the @Data annotation, but didn't go into this much depth, will try understanding into more depth from next time.